File: tests/07_phases_hlt.dg/success_normalize_02.cpp
/*
    The aim of this test is to ensure that induction variables are properly
    adjusted after the normalize HLT transform
*/

/*
<testinfo>
test_generator="config/mercurium-hlt run"
</testinfo>
*/

#include <cassert>


int main()
{
    int i, j;
    unsigned int counter = 0;

    #pragma hlt normalize
    for (i = -10; i < 10; i += 6)
    {
        counter++;
    }

    #pragma hlt normalize
    for (j = 10; j < -10; j -= 6)
    {
        counter++;
    }

    assert(i == 14 && j == -14);
}

File: tests/07_phases_hlt.dg/success_normalize_02.f90
!
! Fortran version of test 'success_normalize_03.cpp'
!

! <testinfo>
! test_generator="config/mercurium-hlt run"
! </testinfo>

PROGRAM P
    IMPLICIT NONE
    INTEGER(1) :: I, J
    INTEGER(4) :: COUNTER

    COUNTER = 0

    !$HLT NORMALIZE
    DO I = -10, 10, 6
        COUNTER = COUNTER + 1
    END DO
    !$HLT END NORMALIZE

    !$HLT NORMALIZE
    DO J = 10, -10, -6
        COUNTER = COUNTER + 1
    END DO
    !$HLT END NORMALIZE

    IF (I .NE. 14 .OR. J .NE. -14) THEN
        WRITE(0, *) "Assertion `I .NE. 14 .OR. J .NE. -14' failed."
        STOP
    END IF
END PROGRAM
